internal/gh/rest.go

```diff
@@ -0,0 +1,219 @@
+    }
+    
+    query := req.URL.Query()
+
+    for key, value := range queryParameters{
+        query.Add(key, value)
+    }
+
+    req.URL.RawQuery = query.Encode()
+
+    return req, nil
+}
+
+//  Update an existing HTTP request using the parameters, token and method.
+func updateHttpRequest(reqUrl, requestMethod string, requestBody io.ReadCloser, queryParameters, headerParameters map[string] string, req *http.Request) error{
+
+    parsedUrl, err := url.Parse(reqUrl)
+    if err != nil {
+        slog.Default().Error("Error while parsing the URL", err)
+	    return err
+    }
+
+    query := parsedUrl.Query()
+    req.Method = requestMethod
+    req.Body = requestBody
+
+    for key, value := range headerParameters{
+        req.Header.Set(key, value)
+    }
+
+    for key, value := range queryParameters{
+        if query.Has(key){
+            query.Set(key, value)
+        } else {
+            query.Add(key, value)
+        }
+        
+    }
+
+    req.URL.RawQuery = query.Encode()
+
+    return nil
+}
+
+//  Entry point for the Force Push processing, setting the url, query paramenters and creating the client.
+func GetForcePushInfo(owner, repo, token, branch string) (int, error){
+
+    slog.Default().Info("Getting repo info - getForcePushInfo method")
+    repoActivityUrl := baseURL + "/repos/" + owner + "/" + repo + "/activity"
+
+    queryParameters := map[string]string{
+        "per_page" : "100",
+        "activity_type" : "force_push",
+        "ref" : branch,
+    }
+
+    headerParameters := map[string]string{
+        "Content-Type" : "application/json",
+        "Authorization": "Bearer "+token,
+    }
+
+    client := &http.Client{}
+
+    NumberForcePush, err := processForcePushRequest(client, repoActivityUrl, queryParameters, headerParameters) 
+    if err != nil {
+        slog.Default().Error("Getting repo info falied - %v getForcePushInfo method", err)
+        return 0, err
+    }    
+    return NumberForcePush, nil
+
+}
+
+//  Execute the HTTP request and return response
+func executeHTTPRequest(client *http.Client, req *http.Request) (*http.Response, error){
+
+    resp, err := client.Do(req)
+	RequestCounter++
+	if err != nil {
+        return nil, err
+	}
+    return resp, nil
+
+}
+
+//  Handle calls to other methods (createHTTPRequest, executeHTTPRequest, processHttpResponse) is handled and loop is added so that requests are processed till there no rel = "next"
+func processForcePushRequest(client *http.Client, repoActivityUrl string, queryParameters, headerParameters map[string] string) (int, error) {
+
+    NumberForcePush := 0
+    hasNext := true
+
+    // Create an initial HTTP Request and set header and query parameters
+    httpReq, err := createHttpRequest(client, repoActivityUrl,"GET", nil, queryParameters, headerParameters)
+    if err != nil {
+        slog.Default().Error("Failed to create HTTP request: %v", err)
+        return  0, err
+    }
+
+    for hasNext{
```

Comment:
run `go fmt` at some point

```suggestion
    for hasNext {
```